dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/event/RumEventSourceResolverTest.kt

```diff
@@ -0,0 +1,278 @@
+    }
+
+    @Test
+    fun `M return null W resolveErrorEventSource { unknown source }`(forge: Forge) {
+        // Given
+        val fakeSourceName = forge.aStringNotMatchingEnum(ErrorEvent.ErrorEventSource::class.java)
+
+        // Then
+        assertThat(testedRumEventSourceResolver.resolveErrorEventSource(fakeSourceName)).isNull()
+    }
+
+    @Test
+    fun `M send an error dev log W resolveErrorEventSource { unknown source }`(forge: Forge) {
+        // Given
+        val fakeSourceName = forge.aStringNotMatchingEnum(ErrorEvent.ErrorEventSource::class.java)
+
+        // When
+        testedRumEventSourceResolver.resolveErrorEventSource(fakeSourceName)
+
+        // Then
+        verify(mockDevLoggerHandler).handleLog(
+            eq(Log.ERROR),
+            eq(
+                RumEventSourceResolver.UNKNOWN_SOURCE_WARNING_MESSAGE_FORMAT
+                    .format(Locale.US, fakeSourceName)
+            ),
+            argThat { this is IllegalArgumentException },
+            eq(emptyMap()),
+            eq(emptySet()),
+            eq(null)
+        )
+    }
+
+    // endregion
+
+    // region Resource
+
+    @Test
+    fun `M resolve the ResourceEvent source W resolveResourceEventSource`(forge: Forge) {
+        // Given
+        val fakeSource = forge.aValueFrom(ResourceEvent.Source::class.java)
+
+        // Then
+        assertThat(testedRumEventSourceResolver.resolveResourceEventSource(fakeSource.name))
+            .isEqualTo(fakeSource)
+    }
+
+    @Test
+    fun `M return null W resolveResourceEventSource { unknown source }`(forge: Forge) {
+        // Given
+        val fakeSourceName = forge.aStringNotMatchingEnum(ResourceEvent.Source::class.java)
+
+        // Then
+        assertThat(testedRumEventSourceResolver.resolveResourceEventSource(fakeSourceName)).isNull()
+    }
+
+    @Test
+    fun `M send an error dev log W resolveResourceEventSource { unknown source }`(forge: Forge) {
+        // Given
+        val fakeSourceName = forge.aStringNotMatchingEnum(ResourceEvent.Source::class.java)
+
+        // When
+        testedRumEventSourceResolver.resolveResourceEventSource(fakeSourceName)
+
+        // Then
+        verify(mockDevLoggerHandler).handleLog(
+            eq(Log.ERROR),
+            eq(
+                RumEventSourceResolver.UNKNOWN_SOURCE_WARNING_MESSAGE_FORMAT
+                    .format(Locale.US, fakeSourceName)
+            ),
+            argThat { this is IllegalArgumentException },
+            eq(emptyMap()),
+            eq(emptySet()),
+            eq(null)
+        )
+    }
+
+    // endregion
+
+    // region Action
+
+    @Test
+    fun `M resolve the ActionEvent source W resolveActionEventSource`(forge: Forge) {
+        // Given
+        val fakeSource = forge.aValueFrom(ActionEvent.Source::class.java)
+
+        // Then
+        assertThat(testedRumEventSourceResolver.resolveActionEventSource(fakeSource.name))
+            .isEqualTo(fakeSource)
+    }
```

Comment:
For enums generated from json schema, the `name`/`valueOf()` rely on the identifier, which is always going to be UPPERCASE_WITH_CAMEL. What we want to rely on is the `jsonValue`. 

In your case, you're testing that `resolveActionEventSource("REACT_NATIVE")` returns `ActionEvent.Source.ANDROID`. What you want to test is resolveActionEventSource("react-native").

Instead of doing `forge.aValueFrom(ActionEvent.Source::class.java)`, you should probably use 
`forge.anElementFrom("android", "ios", "browser", "flutter", "react-native")`

Reply:
oh my bad...I actually wanted to use the json value... 🤦